Amazon.IonObjectMapper/IonObjectSerializer.cs

```diff
@@ -20,7 +20,7 @@ public IonObjectSerializer(IonSerializer ionSerializer, IonSerializationOptions
             this.options = options;
             this.targetType = targetType;
             this.readOnlyProperties = new Lazy<IEnumerable<PropertyInfo>>(
-                () => this.targetType.GetProperties().Where(IsReadOnlyProperty));
+                () => this.targetType.GetRuntimeProperties().Where(IsReadOnlyProperty).Where(HasValidAccessModifier));
```

Comment:
Nit: Let's swap the order of the `Where` conditions.

Reply:
Done